### PR TITLE
[Internal] Tests: Add coverage report

### PR DESCRIPTION
**Pull request by ealsur**
This PR modifies the YAML that runs the Unit Tests project to collect Coverage metrics.

Reference:

* https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/dotnet-core?view=azure-devops#collect-code-coverage
* https://docs.microsoft.com/en-us/azure/devops/pipelines/test/codecoverage-for-pullrequests?view=azure-devops
* https://github.com/MicrosoftDocs/codecoverage-yaml-samples

The format is using Coverlet/Cobertura.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by ealsur** [comment]
Please don't merge this PR, I will merge it myself once the coverage results are visible.

**Comment by ealsur** [comment]
Coverage reports appear on Pipelines, but not yet here https://cosmos-db-sdk-public.visualstudio.com/cosmos-db-sdk-public/_build/results?buildId=19313&view=codecoverage-tab

File: Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangeInternal.cs [hunk 1]
```diff
@@ -47,5 +47,10 @@ internal abstract class FeedRangeInternal : FeedRange
                 return false;
             }
         }
+
+        public static FeedRange Test()
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangeInternal.cs:51]: Why is it needed?

**Comment by kirankumarkolli** [comment]
Kudos. This is awesome. 

Few clarifications
1. Are the results aggregated of all the tests? (UT, emulator etc...)
2. Next step: Automated regression in gates for new PR.

**Comment by ealsur** [comment]
@kirankumarkolli The idea is that only UTs are included, but this is a work in progress, so it might be the rest too.
Currently trying to show the coverage results in GitHub UI

File: Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangeInternal.cs (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangeInternal.cs:51]: This is not needed, its a test to see if the coverage report is on just the PR changes or the whole project.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/FeedRange/FeedRanges/FeedRangeInternal.cs:51]: That is why I added the DO NOT MERGE label 😄

**Comment by ealsur** [comment]
End result is a coverage report on the Checks tab:
image

It also provides the link (title) to the coverage report.

Ideally we will want to track the Diff for the current PR, will track that as a separate workitem.

**Pull request merged**